applications/ConstitutiveLawsApplication/custom_constitutive/hyper_elastic_simo_taylor_neo_hookean_3d.h

```diff
@@ -0,0 +1,256 @@
+// KRATOS ___                _   _ _         _   _             __                       _
+//       / __\___  _ __  ___| |_(_) |_ _   _| |_(_)_   _____  / /  __ ___      _____   /_\  _ __  _ __
+//      / /  / _ \| '_ \/ __| __| | __| | | | __| \ \ / / _ \/ /  / _` \ \ /\ / / __| //_\\| '_ \| '_  |
+//     / /__| (_) | | | \__ \ |_| | |_| |_| | |_| |\ V /  __/ /__| (_| |\ V  V /\__ \/  _  \ |_) | |_) |
+//     \____/\___/|_| |_|___/\__|_|\__|\__,_|\__|_| \_/ \___\____/\__,_| \_/\_/ |___/\_/ \_/ .__/| .__/
+//                                                                                         |_|   |_|
+//
+//  License:         BSD License
+//                   license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+
+#pragma once
```

Comment:
I am still not happy with this because it is inconsistent with the rest of files. I would agree if we decide to update the rest of files in order to be consistent

Reply:
@roigcarlo

Reply:
I know is not standard but so far and to the best of my knowledge the only "real" compiler not supporting it is IAR.

I personally prefer `#pragma` because:
- It's usually more optimal than preprocessing the defines
- Removes the possibility of redefining wards
- Its less prone to error (ie, forget to undefine a macro)

The only known problem using that is for files with the same name (and for the same name I mean their full path) which can be forced used virtual mount points, and its something that we should not support in any case.

To be honest I don't remember if we reached a consensus in the @KratosMultiphysics/technical-committee, but I see not reason not to use it or start changes the old wards when we modify old files (I would not change them all at once because monster PR)

Reply:
i added a card for discussion in the @KratosMultiphysics/technical-committee 

do you have any clue about why they did not make it standard?

Reply:
https://en.wikipedia.org/wiki/Pragma_once#Caveats

Reply:
To be discussed in @KratosMultiphysics/technical-committee . Removed from this PR.